### PR TITLE
death to sleep procs MK I

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Back end code clean up to use call back timers over sleep in some commonly used things.
Makes code look better!

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Wait for Niko to commit on this

**Comment by nikothedude** [comment]
blessed for replacing a lot of 0s and 1s with FALSE and TRUE btw

**Comment by nikothedude** [comment]
blessed, actually, for also removing a bunch of src. and proper return formatting

File: code/game/objects/structures/burrows.dm [hunk 1]
```diff
@@ -639,12 +639,14 @@ percentage is a value in the range 0..1 that determines what portion of this mob
 /****************************
 	Burrow entering
 ****************************/
-/obj/structure/burrow/proc/enter_burrow(var/mob/living/L)
+/obj/structure/burrow/proc/enter_burrow(mob/living/L)
 	break_open()
 	spawn()
```
Comment (nikothedude) [inline code/game/objects/structures/burrows.dm:644]: missed a spawn

File: code/game/objects/structures/satellite.dm [hunk 2]
```diff
@@ -8,8 +8,10 @@
 	if(cooldown == FALSE)
 		cooldown = TRUE
 		emp_in(src.loc, 4, 6, 0)
-		sleep(360)
-		cooldown = FALSE
+		addtimer(CALLBACK(src, .proc/cool_dowed), 360)
```
Comment (nikothedude) [inline code/game/objects/structures/satellite.dm:11]: whats this proc name even mean

File: code/modules/client/edge_sliding/move_loop.dm [hunk 3]
```diff
@@ -61,10 +61,10 @@
 	set waitfor = 0
 	if(src.mloop) return
 	mloop = 1
-	src.Move(get_step(mob.loc, true_dir),true_dir)
-	while(src.true_dir)
+	Move(get_step(mob.loc, true_dir),true_dir)
+	while(true_dir)
 		sleep(world.tick_lag)
```
Comment (nikothedude) [inline code/modules/client/edge_sliding/move_loop.dm:66]: another missed sleep

File: code/modules/mob/living/carbon/superior_animal/defense.dm [hunk 4]
```diff
@@ -245,8 +245,6 @@ mob/living/carbon/superior_animal/adjustToxLoss(var/amount)
 	if (!anim)
 		anim = 0
 
-	sleep(1)
```
Comment (nikothedude) [inline code/modules/mob/living/carbon/superior_animal/defense.dm:248]: whyd ya remove this sleep

File: code/game/objects/structures/burrows.dm (hunk 1 above)
Comment (nikothedude) [reply code/game/objects/structures/burrows.dm:644]: though its important to understand the defualt value of spawn is 0, so this isnt actually sleeping as much as it just waiting for other code to execute

File: code/modules/mob/living/carbon/superior_animal/defense.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/mob/living/carbon/superior_animal/defense.dm:248]: Has no use

File: code/modules/client/edge_sliding/move_loop.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/modules/client/edge_sliding/move_loop.dm:66]: I cant remove that sleep, going to port ssmovement form tg for that

File: code/game/objects/structures/satellite.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/game/objects/structures/satellite.dm:11]: when cooled down

**Pull request merged**